### PR TITLE
S3 uploads

### PR DESCRIPTION
**Pull request by hamishmack**

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# TODO: the file should not be longer needed if we succeed to rely only on
```
Comment (yvan-sraka) [inline .github/workflows/gh-pages.yml:1]: Is this file still needed?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# FIXME: this file should not be longer needed if we succeed to rely only on
```
Comment (yvan-sraka) [inline .github/workflows/main.yml:1]: Is this file still needed?

File: gh-pages/bootstrap.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#! /bin/bash
+set -euo pipefail
+
+# FIXME: this file should not be longer needed if we succeed to rely only on
```
Comment (yvan-sraka) [inline gh-pages/bootstrap.sh:4]: Is this file still needed?

File: .github/workflows/hello.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Nix with good defaults
+        uses: cachix/install-nix-action@v17
+        with:
+          extra_nix_config: |
+            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= loony-tools:pr9m4BkM/5/eSTZlkQyRt57Jz7OMBxNSUiMC4FkcNfk= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+            substituters = https://cache.iog.io/ https://cache.zw3rk.com/ https://cache.nixos.org/
+          nix_path: nixpkgs=channel:nixos-unstable
+      - name: Import DevX closure
```
Comment (yvan-sraka) [inline .github/workflows/hello.yml:19]: Is the purpose of this file to ultimately be a test of the speed download a closure hack shell script and so to hide that behind https://ci.zw3rk.com/job/input-output-hk-devx/pullrequest-38/ghc8107-static-minimal-closure.x86_64-linux/latest/download/2 (e.g., if 2 is the number that refers to `bootstrap.sh` script?)

**Comment by angerman** [comment]
Superseded by #39
